src/js/libcs/OpDrawing.js

```diff
@@ -393,385 +393,339 @@ evaluator.drawconic$1 = function(args, modifs) {
     return eval_helper.drawconic(Conic.matrix, modifs);
 };
 
-eval_helper.drawconic = function(conicMatrix, modifs) {
+// See also eval_helper.quadratic_roots for the complex case
+// Returns either null (if solutions would be complex or NaN)
+// or two pairs [x, y] satisfying ax^2 + bxy + cy^2 = 0
+function solveRealQuadraticHomog(a, b, c) {
+    var d = b * b - 4 * a * c;
+    /*jshint -W018 */
+    if (!(d >= 0)) return null; // also return null if d is NaN
+    /*jshint +W018 */
+    var r = Math.sqrt(d);
+    if (b > 0) r = -r;
+    return [
+        [r - b, 2 * a],
+        [2 * c, r - b]
+    ];
+}
+
+// Returns either null (if solutions would be complex or NaN)
+// or two values x satisfying ax^2 + bx + c = 0
+function solveRealQuadratic(a, b, c) {
+    var hom = solveRealQuadraticHomog(a, b, c);
+    if (hom === null) return null;
+    return [hom[0][0] / hom[0][1], hom[1][0] / hom[1][1]];
+}
+
+function inRange(x, a, b) {
+    return x >= Math.min(a, b) && x <= Math.max(a, b);
+}
 
+eval_helper.drawconic = function(conicMatrix, modifs) {
     Render2D.handleModifs(modifs, Render2D.conicModifs);
     if (Render2D.lsize === 0)
         return;
     Render2D.preDrawCurve();
 
-    var eps = 1e-14; //JRG Hab ih von 1e-16 runtergesetzt
+    var eps = 1e-14;
     var mat = List.normalizeMax(conicMatrix);
-    var origmat = mat;
-
-    // check for complex values
-    for (var i = 0; i < 2; i++)
-        for (var j = 0; j < 2; j++) {
-            if (Math.abs(mat.value[i].value[j].value.imag) > CSNumber.eps) return;
+    var tmat = csport.toMat();
+    mat = List.mult(List.transpose(tmat), mat);
+    mat = List.mult(mat, tmat);
+    mat = List.normalizeMax(mat);
+
+    // Using polynomial coefficients instead of matrix
+    // since it generalizes to higher degrees more easily.
+    // cij is the coefficient of the monomial x^i * y^j.
```

Comment:
We should add some checking for complex values here, otherwise 

```
C0 = [-0.31, -0.27, -0.13, 2.8, 0.09, 1];
C3 = [-0.31+i, -0.27, -0.13+i, 2.8, 0.09, 1];
``` 

will be drawn as the same conic. Choosing the epsilon here should not be too strict, otherwise we run into a blinking Christmas tree of disappearing and reappearing circles.